### DISCUSSION TITLE
Specifying UTF-8 encoding?

### DISCUSSION THREAD

**Discussion by codeconscious** [statiqdev/Discussions#91] [Q&A] [answered]
I've just started trying out this tool today and am impressed thus far. There's one issue I haven't been able to work out yet on my own or by looking through the documentation: How can I specify UTF-8 encoding for either the entire site or else, if that's not possible, by page? I want to type Japanese in my markdown files, but it's not rendering correctly.

**Comment by TheBrambleShark** [comment, marked as answer]
You should be able to specify this in your head. Add this to your _MasterLayout.cshtml
```html
<head>
  <meta charset="utf-8">
</head>
```

**Reply by codeconscious** [reply]
Thank you for your reply. Sorry if I'm missing something obvious, but I don't have this file in my project and creating it manually with the HTML above didn't seem to do anything no matter where I placed the file.

Incidentally, I originally created the project with the following command:
`dotnet add package Statiq.Web --version 1.0.0-beta.33`

Might there be something I'm overlooking?

**Reply by TheBrambleShark** [reply]
@daveaglick Should we update the tutorial on https://www.statiq.dev/web/ to suggest users create a new project using `dotnet new web` or `dotnet new webapp` (asp.net core empty or asp.net core web app, respectively) so they can take advantage of the layout documents et al?

> Might there be something I'm overlooking?

I don't think so. If I had to guess, it sounds like you're developing against a standard console app instead of an asp.net core web app.

I would try creating that `_MasterLayout.cshtml` file inside the root of your input directory. Exact name is important. This will be the file where you set up anything that is common across all pages, such as your html head and perhaps some layout components. Mine is a little non-standard, but if you'd like an example of one working with Statiq, you can view that here:

https://github.com/LuzFaltex/luzfaltex.com/blob/main/input/_MasterLayout.cshtml

**Reply by daveaglick** [reply]
@Foxtrek64 Thanks for the assistance on this one!

@codeconscious You're doing it fine :). The "quick start" instructions on https://www.statiq.dev/web/ get you up and running with a console app, but we don't want to make any assumptions about your layout, so the "default" site is fairly simple - it has support for things like blog posts but no layout files (like what @Foxtrek64 referred to).

A lot of folks end up either creating their own layouts using Razor or make use of an entire theme like [CleanBlog](https://github.com/statiqdev/CleanBlog) (this is the only pre-canned theme at the moment, but more are planned). In both cases, the file being used for the layout can be specified as a [setting](https://www.statiq.dev/web/configuration/settings), either in a `settings.yml` file or in code by using `AddSetting()`:

`settings.yml`
```
Layout: /_layout.cshtml
```

The layout file is where you'd probably want to add the `meta` element in the `<head>` to specify a UTF-8 page. Everything else in that layout file will also be "inherited" by the pages in your site. You can also specify a different layout file name (which is where @Foxtrek64 is getting `_MasterLayout.cshtml` from - that's the layout his own site or theme is using).

The CleanBlog theme already has the UTF-8 `meta` element, so you can see it and what the rest of a Razor layout file looks like at  https://github.com/statiqdev/CleanBlog/blob/main/input/_layout.cshtml

**Reply by codeconscious** [reply]
Thank you both very much for the information and your patience! I really appreciate it. I'll look into everything a bit more and work out a layout that works for me. Thanks again.
